Add multiple algorithm support in compute_igraph

sky-2002 opened the pull request:
@silkspace , This PR tries to solve issue mentioned [here](https://github.com/graphistry/pygraphistry/issues/405). With this, multiple algorithms can be passed as a list to the `compute_igraph` function.

lmeyerov commented:
@sky-2002  see my comment on https://github.com/graphistry/pygraphistry/issues/405 wrt more usable signature

looks like lint/type errors, see ci . you can build & test locally

see https://github.com/graphistry/pygraphistry/blob/master/DEVELOP.md#run-local-tests-without-rebuild + look at the script for env vars you can set

sky-2002 commented:
Hello @lmeyerov , @silkspace , I have made the necessary changes for more usable signature, added code to support custom fieldnames as suggested by @DataBoyTX in #405 and added tests too. Let me know if anything needs to be changed.

lmeyerov commented on graphistry/tests/plugins/test_igraph.py:
```diff
@@ -516,6 +516,16 @@ def test_all_calls(self):
+        multiple1 = compute_igraph(g,{'pr':('pagerank',{'damping': 0.85}),
+                                     'coci':'cocitation','cl':'closeness','bt':'betweenness'})
+        assert 'cl' in multiple1._nodes.columns
+        assert 'bt' in multiple1._nodes.columns
```
assert also `pr` & `coci`

lmeyerov commented on graphistry/tests/plugins/test_igraph.py:
```diff
@@ -516,6 +516,16 @@ def test_all_calls(self):
+        assert 'bt' in multiple1._nodes.columns
+
+        multiple2 = compute_igraph(g, [('pagerank',{'damping': 0.85}),'closeness'])
+        assert 'closeness' in multiple2._nodes.columns
```
assert also `pagerank`

lmeyerov commented on graphistry/plugins/igraph.py:
```diff
@@ -282,7 +282,8 @@ def to_igraph(self: Plottable,
 def compute_igraph(
     self: Plottable,
-    alg: str,
+    # alg: str,
```
remove

lmeyerov commented on graphistry/plugins/igraph.py:
```diff
@@ -282,7 +282,8 @@ def to_igraph(self: Plottable,
     self: Plottable,
-    alg: str,
+    # alg: str,
+    alg: Union[str, List[Union[str, Tuple[str, dict]]], Dict[str, Union[str, Tuple[str,dict]]] ],
```
Add `** Example...` of the new forms

lmeyerov reviewed (changes requested):
Getting there!

* See CI errors
* Can you add an example to `readme.md` of the new forms?
* Can you add to the changelog? 
* See inline review comments

sky-2002 commented:
> Getting there!
> 
> * See CI errors
> * Can you add an example to `readme.md` of the new forms?
> * Can you add to the changelog?
> * See inline review comments

For the changelog, will this be fine - "Multiple algorithm support for igraph" in the [Development] Added section.

lmeyerov commented:
Looks like the type checker needs help?

sky-2002 commented:
> Looks like the type checker needs help?

Well, tried to fix it, let's see.

lmeyerov commented:
Great. Did you try running the tests locally? Set the various env vars of interest to `=1` @ https://github.com/graphistry/pygraphistry/blob/master/DEVELOP.md#run-local-tests-without-rebuild

lmeyerov commented:
We can revisit this when more cycles, closing for now
